### PR TITLE
#Vulcan 134/ordering and hide features of attributes in node/edge detailed modal

### PR DESCRIPTION
**Pull request by m-o-n-i-s-h**
ordering and hide features of attributes in node/edge detailed modal

### CODE REVIEW

**Comment by m-o-n-i-s-h** [comment]
```
"customizedOrderingOfAttributesInDetailView": [
  {
    "type": "node",
    "entityType": "NTG6_HU_SW",
    "ordering": ["title", "sgenComment", "release", "notes"],
     "hide": ["release"]
     }
]
```

**Review by brahmprakashMishra — APPROVED** [review]

**Comment by brahmprakashMishra** [comment]
@m-o-n-i-s-h  some merge conflicts appearing - could you resolve?

File: src/config/ReportConfig.tsx [hunk 1]
```diff
@@ -349,6 +349,12 @@ export const REPORT_TYPES = {
         values: [true, false],
         default: false,
       },
+      customizedOrderingOfAttributesInDetailView: {
```
[the remark this answers is not in the document]
Comment (m-o-n-i-s-h) [reply src/config/ReportConfig.tsx]: changed

File: src/chart/graph/component/GraphChartInspectModal.tsx [hunk 2]
```diff
@@ -8,6 +8,13 @@ import GraphEntityInspectionTable from './GraphEntityInspectionTable';
  * Renders a pop-up window to inspect a node/relationship properties in a read-only table.
  */
 export const NeoGraphChartInspectModal = (props: GraphChartVisualizationProps) => {
+  const tableDataCustomSettings = Array.isArray(props.interactivity?.customizedOrderingOfAttributesInDetailView)
+    ? props.interactivity?.customizedOrderingOfAttributesInDetailView?.find(
+        (setting) =>
+          setting.entityType ===
+          (props.interactivity.selectedEntity ? getEntityHeader(props.interactivity?.selectedEntity) : '')
```
[the remark this answers is not in the document]
Comment (m-o-n-i-s-h) [reply src/chart/graph/component/GraphChartInspectModal.tsx]: done

File: src/chart/graph/component/GraphEntityInspectionTable.tsx [hunk 3]
```diff
@@ -18,9 +18,16 @@ export const GraphEntityInspectionTable = ({
...
   },
   checklistEnabled = false,
+  customizedOrderingOfAttributesInDetailView,
 }) => {
   const [checkedParameters, setCheckedParameters] = React.useState<string[]>([]);
   const hasPropertyToShow = Object.keys(entity.properties).length > 0;
+  const orderedAttributeList = customizedOrderingOfAttributesInDetailView?.ordering || [];
+  const unOrderedAttributeList =
+    Object.keys(entity.properties).filter(
+      (p) => !(customizedOrderingOfAttributesInDetailView?.ordering || []).includes(p)
```
[the remark this answers is not in the document]
Comment (m-o-n-i-s-h) [reply src/chart/graph/component/GraphEntityInspectionTable.tsx]: Done

File: src/chart/graph/component/GraphEntityInspectionTable.tsx [hunk 4]
```diff
@@ -18,9 +18,16 @@ export const GraphEntityInspectionTable = ({
     console.log('undefined function in GraphEntityInspectionTable');
   },
   checklistEnabled = false,
+  customizedOrderingOfAttributesInDetailView,
 }) => {
   const [checkedParameters, setCheckedParameters] = React.useState<string[]>([]);
   const hasPropertyToShow = Object.keys(entity.properties).length > 0;
+  const orderedAttributeList = customizedOrderingOfAttributesInDetailView?.ordering || [];
+  const unOrderedAttributeList =
```
[the remark this answers is not in the document]
Comment (m-o-n-i-s-h) [reply src/chart/graph/component/GraphEntityInspectionTable.tsx]: Done

File: src/chart/graph/GraphChart.tsx [hunk 5]
```diff
@@ -204,6 +204,7 @@ const NeoGraphChart = (props: ChartProps) => {
       clickPosition: clickPosition,
       setClickPosition: setClickPosition,
       createNotification: props.createNotification,
+      customizedOrderingOfAttributesInDetailView: settings.customizedOrderingOfAttributesInDetailView,
```
[the remark this answers is not in the document]
Comment (m-o-n-i-s-h) [reply src/chart/graph/GraphChart.tsx]: Done

**Pull request merged**